README.adoc

```diff
@@ -2,188 +2,184 @@
-[source,bash]
+[source,java]
 ----
-./mvnw spring-boot:run -pl eureka-service
+include::servicea/src/main/java/com/example/servicea/controller/ServiceARestController.java[]
 ----
 ====
 
-To run the Eureka client with Maven, run the following command in a terminal window (in
-the `/complete` directory):
+== Service B Endpoint
+
+Create a new class `com/example/serviceb/controller/ServiceBRestController.java` in the `serviceb` project that exposes an endpoint that calls `servicea`.
 
 ====
-[source,bash]
+[source,java]
 ----
-./mvnw spring-boot:run -pl eureka-client
+include::serviceb/src/main/java/com/example/serviceb/controller/ServiceBRestController.java[]
 ----
 ====
 
-To run the Eureka service with Gradle, run the following command in a terminal window (in
-the `/complete` directory):
+This class uses a `DiscoveryClient` to find the `ServiceInstance` of `servicea` based only on the application name.
+This guide only has one instance of `servicea`, so we can look only at the first instance.
+This is seen in the line:
+
+```
+ServiceInstance serviceInstance = discoveryClient.getInstances("servicea").get(0);
+```
+
+Once you have the `ServiceInstance` that refers to the location of `servicea`, you can now use the information in Spring Framework's https://docs.spring.io/spring-framework/reference/integration/rest-clients.html#rest-restclient[Rest Client].
+
+```
+
+... = restClient.get()
+				.uri(serviceInstance.getUri() + "/helloWorld")...
+
+```
+
+You can test all three applications by executing the command:
 
 ====
 [source,bash]
 ----
-./gradlew :eureka-service:bootRun
+curl http://localhost:8082/serviceB/helloEureka
 ----
 ====
 
-To run the Eureka client with Gradle, run the following command in a terminal window (in
-the `/complete` directory):
+And you should see the result:
 
 ====
 [source,bash]
 ----
-./gradlew :eureka-client:bootRun
+Hello world from Service A!
 ----
 ====
 
-The `eureka-client` will take about a minute to register itself in the registry and to
-refresh its own list of registered instances from the registry. Visit the `eureka-client`
-in the browser, at `http://localhost:8080/service-instances/a-bootiful-client`. There, you
-should see the `ServiceInstance` for the `eureka-client` reflected in the response. If you
-see an empty `<List>` element, wait a bit and refresh the page.
+== Test the Application
+
+When you have implemented the code above, or you are using the solution repository, here is an outline of the steps in this guide:
+
+- Run `eureka-server` with the command `./gradlew bootRun` from the `eureka-server` folder
+- Run `servicea` with the command `./gradlew bootRun` from the `servicea` folder
+- Run `serviceb` with the command `./gradlew bootRun` from the `serviceb` folder
+- Observe `servicea` and `serviceb` are registered by viewing the Eureka Dahsboard at http://localhost:8761/
+- Run the command `curl http://localhost:8082/serviceB/helloEureka` to test all three applications are working properly
```

Comment:
Change "command `curl http://localhost:8082/serviceB/helloEureka`" to "`curl http://localhost:8082/serviceB/helloEureka` command"